src/go/MONIT/ggus_alerting.go

```diff
@@ -0,0 +1,357 @@
+			State       string        `json:"state"`
+			SilencedBy  []interface{} `json:"silencedBy"`
+			InhibitedBy []interface{} `json:"inhibitedBy"`
+		} `json:"status"`
+		Receivers   []string `json:"receivers"`
+		Fingerprint string   `json:"fingerprint"`
+	} `json:"data"`
+}
+
+//function for parsing JSON data from GGUS Data
+func (data *ggus) parseJSON(jsondata []byte) {
+	err := json.Unmarshal(jsondata, &data)
+	if err != nil {
+		log.Printf("Unable to parse GGUS JSON data, error: %v\n", err)
+	}
+}
+
+//function for fetching JSON data from GGUS Data
+func fetchJSON(filename string) []byte {
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Printf("Unable to open JSON file, error: %v\n", err)
+	}
+
+	defer file.Close()
+
+	jsonData, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Printf("Unable to read JSON file, error: %v\n", err)
+	}
+
+	if verbose > 1 {
+		log.Println("GGUS Data: " + string(jsonData))
+	}
+
+	return jsonData
+
+}
+
+//function for converting SSB JSON Data into JSON Data required by AlertManager APIs.
+func convertData(data ggus) []byte {
+
+	var temp amJSON
+	var finalData []amJSON
+
+	exstTkt = make(map[string]int)
+
+	for _, each := range data {
+
+		// Ignoring all tickets whose VO is not "cms"
+		if each.VO != "cms" {
+			continue
+		}
+
+		exstTkt[strconv.Itoa(each.TicketID)] = 1
+
+		begin := int64(each.LastUpdate)
+		_beginRFC3339 := time.Unix(begin, 0).UTC()
+
+		temp.Labels.Alertname = "ggus-" + strconv.Itoa(each.TicketID)
+		temp.Labels.Severity = severity
+		temp.Labels.Tag = tag
+
+		temp.Annotations.TicketID = strconv.Itoa(each.TicketID)
+		temp.Annotations.Type = each.Type
+		temp.Annotations.VO = each.VO
+		temp.Annotations.Site = each.Site
+		temp.Annotations.Priority = each.Priority
+		temp.Annotations.ResponsibleUnit = each.ResponsibleUnit
+		temp.Annotations.Status = each.Status
+		temp.Annotations.Subject = each.Subject
+		temp.Annotations.Scope = each.Scope
+		temp.Annotations.URL = "https://ggus.eu/?mode=ticket_info&ticket_id=" + strconv.Itoa(each.TicketID)
+
+		temp.StartsAt = _beginRFC3339
+
+		finalData = append(finalData, temp)
+
+	}
+
+	jsonStr, err := json.Marshal(finalData)
+
+	if err != nil {
+		log.Printf("Unable to convert JSON Data, error: %v\n", err)
+	}
+
+	return jsonStr
+
+}
+
+//function for get request on /api/v1/alerts alertmanager endpoint for fetching alerts.
+func get() *getAMJSON {
+
+	var data *getAMJSON
+
+	apiurl := alertManagerURL + "/api/v1/alerts?active=true&silenced=false&inhibited=false&unprocessed=false"
+
+	req, err := http.NewRequest("GET", apiurl, nil)
+	req.Header.Add("Accept-Encoding", "identity")
+	req.Header.Add("Accept", "application/json")
+
+	client := &http.Client{}
+
+	if verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
```

Comment:
Will it work for alerts which will have different attributes then ggus ones?

Reply:
Yes ! I have injected both alerts (SSB & GGUS Data) and it works independently. Please let me know if I still have to filter the GGUS data.

Reply:
I have added the filter for fetching only GGUS alerts from AlertManager. 
```
apiurl := alertManagerURL + "/api/v1/alerts?active=true&silenced=false&inhibited=false&unprocessed=false&filter=tag=\"GGUS\""
```